documentation/course/e2/README.md

```diff
@@ -2164,7 +2166,37 @@ TODO @jakubzehner: Dodać diagram pakietów, opis architektury i endpointy.
 
 ## Płatność
 
-TODO @piterek130: Dodać diagram pakietów, opis architektury.
+Do implementacji serwisu Inferius wybrano język **Java** z frameworkiem **Spring Boot**.
+
+Wybór ten podyktowany był następującymi czynnikami:
+
+- **doświadczenie zespołu** - większość członków zespołu posiada doświadczenie w programowaniu w języku Java, co pozwoli na spełnienie wymogu wymienionego wyżej - znajomość języka przez co najmniej jedną osobę spoza serwisu w celu przeprowadzenia code review,
+- **dostępność AWS SDK** - AWS SDK wspiera język Java, co pozwoli na łatwe korzystanie z usług AWS,
+- **dostępność bibliotek** - Java posiada bardzo rozbudowaną bazę bibliotek, co pozwoli na szybkie i efektywne rozwijanie serwisu,
+- **dojrzałość frameworka** - Spring Boot jest jednym z najpopularniejszych frameworków do tworzenia aplikacji w języku Java, co pozwoli na szybkie rozwiązywanie ewentualnych problemów,
+- **łatwość w testowaniu** - Java jest językiem, który posiada wiele rozbudowanych narzędzi do testowania, co ułatwi proces testowania serwisu.
+
+![Diagram pakietów Inferius](./images/package-diagram-inferius.drawio.svg)
+
+Architektura podsystemu Inferius została zaprojektowana zgodnie z podejściem **Vertical Slice Architecture**, które promuje podział aplikacji na niezależne funkcjonalne jednostki. Każda jednostka (slice) jest odpowiedzialna za jeden aspekt działania systemu i obejmuje wszystkie potrzebne warstwy: dostęp do bazy danych, logikę biznesową, API oraz testy.
+
+Każdy slice w systemie odpowiada jednej z kluczowych domen:
+- **Wallet**: Odpowiada za zarządzanie portfelem pasażera, przechowując informacje o jego saldzie.
+- **Fine**: Zarządza mandatami pasażerów, w tym ich szczegółami, powodami wystawienia i statusem płatności.
+- **CreditCardInfo**: Obsługuje dane kart kredytowych pasażerów, takie jak numer karty, dane posiadacza i datę ważności.
+- **WalletHistory**: Rejestruje historię operacji finansowych, takich jak doładowania portfela.
```

Comment:
**thought:** Does `WalletHistory` have to be separate from `Wallet`? They seem to be very close logically.